warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/BaseIngestHelper.java

```diff
@@ -602,16 +614,43 @@ public List<datawave.data.type.Type<?>> getDataTypes(String fieldName) {
                 compilePatterns();
             }
 
+            List<Integer> patternLengths = new ArrayList<>();
+            int bestMatch = 0;
+            Collection<datawave.data.type.Type<?>> bestMatchTypes = null;
+
             for (Matcher patternMatcher : typeCompiledPatternMap.keySet()) {
```

Comment:
Additional suggestions that are for extra credit only:

1) only supply a warning message when we have two or more regexes with the same length that have different normalizers
2) If those two or more regexes are the final "bestMatch", and there are differing normalizers, then supply an error message instead
    2.a) Lets reevaluate whether we should fail in this circumstance